parlai/mturk/tasks/acute_eval/frontend/components/custom.jsx

```diff
@@ -0,0 +1,576 @@
+            backgroundColor: primary_speaker_color
+          }
+        : {
+            textAlign: "right",
+            float: "right",
+            display: "inline-block",
+            marginLeft: 20,
+            backgroundColor: otherspeaker_color
+          })
+    };
+    return (
+      <div style={message_container_style}>
+        <div style={message_style}>{message}</div>
+      </div>
+    );
+  }
+}
+
+class MessageList extends React.Component {
+  makeMessages() {
+    let agent_id = this.props.agent_id;
+    if (this.props.task_data.pairing_dict === undefined) {
+      return (
+        <div>
+          <p> Loading chats </p>
+        </div>
+      );
+    }
+    let task_data = this.props.task_data;
+    let conv_order_ind =
+      task_data.task_specs.conversation_order[this.props.index];
+    let messages =
+      task_data.pairing_dict.dialogue_dicts[conv_order_ind].dialogue;
+    console.log("MESSAGES");
+    console.log(task_data.pairing_dict.dialogue_dicts[conv_order_ind]);
+    console.log(messages);
+    let primary_speaker =
+      task_data.pairing_dict.speakers_to_eval[conv_order_ind];
+
+    let XChatMessage = getCorrectComponent("XChatMessage", this.props.v_id);
+    let onClickMessage = this.props.onClickMessage;
+    if (typeof onClickMessage !== "function") {
+      onClickMessage = idx => {};
+    }
+    return messages.map((m, idx) => (
+      <div key={conv_order_ind + "_" + idx} onClick={() => onClickMessage(idx)}>
+        <XChatMessage
+          message={m.text}
+          is_primary_speaker={m.speaker == primary_speaker}
+          conv_ind={this.props.index}
+        />
+      </div>
+    ));
+  }
+
+  render() {
+    return (
+      <div id="message_thread" style={{ width: "100%" }}>
+        {this.makeMessages()}
+      </div>
+    );
+  }
+}
+
+class ChatPane extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { chat_height: this.getChatHeight() };
+  }
+
+  getChatHeight() {
+    let entry_pane = $("div#right-bottom-pane").get(0);
+    let bottom_height = 90;
+    if (entry_pane !== undefined) {
+      bottom_height = entry_pane.scrollHeight;
+    }
+    return this.props.frame_height - bottom_height;
+  }
+
+  handleResize() {
+    if (this.getChatHeight() != this.state.chat_height) {
+      this.setState({ chat_height: this.getChatHeight() });
+    }
+  }
+
+  render() {
+    let v_id = this.props.v_id;
+    let XMessageList = getCorrectComponent("XMessageList", v_id);
+
+    // TODO move to CSS
+    let top_pane_style = {
+      width: "100%",
+      position: "relative",
+      overflowY: "scroll"
+    };
+
+    let chat_style = {
+      width: "100%",
+      height: "100%",
+      paddingTop: "60px",
+      paddingLeft: "20px",
+      paddingRight: "20px",
+      paddingBottom: "20px",
+      overflowY: "scroll"
+    };
+
+    window.setTimeout(() => {
+      this.handleResize();
+    }, 10);
+
+    top_pane_style["height"] = this.state.chat_height + "px";
```

Comment:
any reason why this line isn't just part of the `top_pane_style` object above?

Reply:
I don't remember there being a good reason, I can probalby move it next PR